internal/cloud/cloud.go

```diff
@@ -33,4 +33,7 @@ const (
 	TagUID = "constellation-uid"
 	// TagInitSecretHash is the tag/label key used to identify the hash of the init secret.
 	TagInitSecretHash = "constellation-init-secret-hash"
+	// TagCustomEndpoint is the tag/label key used to identify the custom endpoint
+	// or dns name that should be added to tls cert SANs.
+	TagCustomEndpoint = "constellation-custom-endpoint"
```

Comment:
I'm thinking about moving this to `InitRequest` in `init.proto` since we don't need this value pre-init, which is different than the other values here.

Reply:
Yeah that's possible and not a bad solution. I feel like the current approach is cleaner for the consumer of `GetAPIServerCertSANs`, since all fields in the list of SANs can be retrieved from the CSP metadata implementation. Otherwise, the consumer has to combine info from metadata package and init call.
But if you have a strong preference for the init.proto, I am happy to change it.

Reply:
Hm, I think that the most obvious drawback is the limitation set by the CSP. I just checked and all 3 major CSPs support a superset of DNS names. I don't think that there is future in which we support more than one custom DNS name, right?
On the other hand, I think that we'll at some point send over large(r) parts of the Kubernetes config in an attempt to make it configurable. So the list of values we send over from the cli and that are needed in the K8s bootstrapping will increase.

I'm not 100% sure, maybe there's a third person wanting share their opinion?

Reply:
>  I don't think that there is future in which we support more than one custom DNS name, right?

Agreed. I don't see a use case for more than one custom DNS name.

> On the other hand, I think that we'll at some point send over large(r) parts of the Kubernetes config in an attempt to make it configurable.

If we do that, we can also get rid of most of the cloud provider metadata package, no? So it feels to me like we should either send everything via init or try to get everything from csp metadata but avoid splitting this into two sources.

> maybe there's a third person wanting share their opinion?

Maybe @daniel-weisse since you know the init.proto quite well?

Reply:
I don't see any immediate downsides to sending this as part of init.
Also seems a bit more intuitive than setting the DNS names as some metadata tag.

I'm not sure how new nodes receive this value currently. If they also rely on metadata tags, we need to take some care in adjusting that.

Reply:
> I'm not sure how new nodes receive this value currently. If they also rely on metadata tags, we need to take some care in adjusting that.

new nodes receive this value via the kubeadm configmap.

You convinced me. I'll just move this logic to the initproto.